clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -172,7 +172,11 @@ private void initializeTransactionManager(Optional<String> transactionalId, bool
             new ApiVersion()
                 .setApiKey(ApiKeys.PRODUCE.id)
                 .setMinVersion((short) 0)
-                .setMaxVersion((short) 7)),
+                .setMaxVersion(transactionV2Enabled ? (short) 12 : (short) 11),
```

Comment:
nit: should we make this apiKeys.Produce.latestVersion()

(was there any reason why the max version was 7?)